### ISSUE TITLE
Aws::SharedCredentials regression

### ISSUE THREAD

**Issue by temujin9** [aws/aws-sdk-ruby#1257]
v.2.4.0 and v2.4.1 work fine with AWS_PROFILE role assumption. v2.4.2 or later fail 
```
/home/temujin9/.gem/ruby/1.9.1/gems/aws-sdk-core-2.4.2/lib/aws-sdk-core/plugins/request_signer.rb:100:in `require_credentials': unable to sign request without credentials set (Aws::Errors::MissingCredentialsError)
        from /home/temujin9/.gem/ruby/1.9.1/gems/aws-sdk-core-2.4.2/lib/aws-sdk-core/plugins/request_signer.rb:90:in `sign_authenticated_requests'
        from /home/temujin9/.gem/ruby/1.9.1/gems/aws-sdk-core-2.4.2/lib/aws-sdk-core/plugins/request_signer.rb:83:in `call'
        from /home/temujin9/.gem/ruby/1.9.1/gems/aws-sdk-core-2.4.2/lib/aws-sdk-core/plugins/helpful_socket_errors.rb:10:in `call'
        from /home/temujin9/.gem/ruby/1.9.1/gems/aws-sdk-core-2.4.2/lib/aws-sdk-core/plugins/retry_errors.rb:87:in `call'
        from /home/temujin9/.gem/ruby/1.9.1/gems/aws-sdk-core-2.4.2/lib/aws-sdk-core/query/handler.rb:27:in `call'
        from /home/temujin9/.gem/ruby/1.9.1/gems/aws-sdk-core-2.4.2/lib/aws-sdk-core/plugins/user_agent.rb:12:in `call'
        from /home/temujin9/.gem/ruby/1.9.1/gems/aws-sdk-core-2.4.2/lib/seahorse/client/plugins/endpoint.rb:41:in `call'
        from /home/temujin9/.gem/ruby/1.9.1/gems/aws-sdk-core-2.4.2/lib/seahorse/client/plugins/raise_response_errors.rb:14:in `call'
        from /home/temujin9/.gem/ruby/1.9.1/gems/aws-sdk-core-2.4.2/lib/aws-sdk-core/plugins/param_converter.rb:20:in `call'
        from /home/temujin9/.gem/ruby/1.9.1/gems/aws-sdk-core-2.4.2/lib/seahorse/client/plugins/response_target.rb:21:in `call'
        from /home/temujin9/.gem/ruby/1.9.1/gems/aws-sdk-core-2.4.2/lib/seahorse/client/request.rb:70:in `send_request'
        from /home/temujin9/.gem/ruby/1.9.1/gems/aws-sdk-core-2.4.2/lib/seahorse/client/base.rb:207:in `block (2 levels) in define_operation_methods'
        from /home/temujin9/.gem/ruby/1.9.1/gems/cloudformation-ruby-dsl-1.2.1/lib/cloudformation-ruby-dsl/cfntemplate.rb:297:in `cfn'
        from /home/temujin9/.gem/ruby/1.9.1/gems/cloudformation-ruby-dsl-1.2.1/lib/cloudformation-ruby-dsl/cfntemplate.rb:556:in `exec!'
        from ./nexpose-console.rb:158:in `<main>'
```

This has forced us to [pin the version in our Gemfile](https://github.com/bazaarvoice/cloudformation-ruby-dsl/commit/c60fdd002217f2e84240c37d036d658ffd7505ac) to use the shared credentials.

**Comment by awood45** [comment]
Three part answer:

First, the workaround. Make sure that the `AWS_SDK_LOAD_CONFIG` environment variable is set, for example to `"1"`, and your code should work. That's the feature flag environment variable, and setting it will get you back on track.

Second, why this happened in `2.4.2`. The `2.4.0` and `2.4.1` releases had a code bug where we ignored the feature flag and always used the new credential resolution chain, which is why your code worked without a feature flag. Technically, the new credential resolution chain is a breaking change, hence the feature flag.

Third, what might happen. We're discussing if we should remove the feature flag, since the breaking change surface area is low. That discussion is ongoing (and we're open to input).

For now - set the environment variable `AWS_SDK_LOAD_CONFIG` and you'll be working again.

**Comment by temujin9** [comment]
Aha! We had tried that earlier, but in the storm of problems (this was not the only one) we lost track of trying it again.

I strongly vote for the removal of the flag. This is the documented way that aws configuration is supposed to work. Inobvious flags needed in addition end up confusing the issue.

As we have it patched (via version pinning) where we need it for the moment, I think I'm just going to leave it as is until you've decided. Can you ping here when you do?

**Comment by awood45** [comment]
Sure, I can leave this open.

**Comment by awood45** [comment]
#1263 creates an opt-out flag, and makes this behavior on by default. When merged, it will be released as version `2.5.0`.
